db/version_builder.cc

```diff
@@ -551,42 +608,42 @@ class VersionBuilder::Rep {
     auto base_it = base_blob_files.begin();
     const auto base_it_end = base_blob_files.end();
 
-    auto changed_it = changed_blob_files_.begin();
-    const auto changed_it_end = changed_blob_files_.end();
+    auto delta_it = blob_file_meta_deltas_.begin();
+    const auto delta_it_end = blob_file_meta_deltas_.end();
 
-    while (base_it != base_it_end && changed_it != changed_it_end) {
+    while (base_it != base_it_end && delta_it != delta_it_end) {
       const uint64_t base_blob_file_number = base_it->first;
-      const uint64_t changed_blob_file_number = changed_it->first;
+      const uint64_t delta_blob_file_number = delta_it->first;
 
-      const auto& base_meta = base_it->second;
-      const auto& changed_meta = changed_it->second;
-
-      assert(base_meta);
-      assert(changed_meta);
-
-      if (base_blob_file_number < changed_blob_file_number) {
+      if (base_blob_file_number < delta_blob_file_number) {
+        const auto& base_meta = base_it->second;
+        assert(base_meta);
         assert(base_meta->GetGarbageBlobCount() <
                base_meta->GetTotalBlobCount());
 
         vstorage->AddBlobFile(base_meta);
 
         ++base_it;
-      } else if (changed_blob_file_number < base_blob_file_number) {
-        AddBlobFileIfNeeded(vstorage, changed_meta);
+      } else if (delta_blob_file_number < base_blob_file_number) {
```

Comment:
Question:
If it's a new blob file, will it ever be smaller than the blob files already in base storage version?

Reply:
Good point. No, with ever-increasing blob file numbers it is not possible to hit this branch. I was thinking about keeping this method fully generic but I guess it does not make much sense since we (will) rely on the strictly increasing property elsewhere. Will remove this and add an assertion.